aep/general/0134/aep.md.j2

```diff
@@ -0,0 +1,321 @@
+# Update
+
+In REST APIs, it is customary to make a `PATCH` or `PUT` request to a
+resource's URI (for example, `/v1/publishers/{publisher}/books/{book}`) in
+order to update that resource.
+
+Resource-oriented design (AEP-121) honors this pattern through the `Update`
+method (which mirrors the REST `PATCH` behavior). These RPCs accept the URI
+representing that resource and return the resource.
+
+## Guidance
+
+APIs **should** provide an update method for resources unless it is not
+valuable for users to do so. The purpose of the update method is to make
+changes to the resources without causing side effects.
+
+Update methods are specified using the following pattern:
+
+- The RPC's name **must** begin with the word `Update`. The remainder of the
+  RPC name **should** be the singular form of the resource's message name.
+- The request message **must** exactly match the RPC name, with a `Request`
+  suffix.
+- The response message **must** be the resource itself.
+  - The response **should** include the fully-populated resource, and **must**
+    include any fields that were sent and included in the update mask unless
+    they are input only (see AEP-203).
+  - If the update RPC is [long-running](#long-running-update), the response
+    message **must** be an `Operation` for which the return type is the
+    resource itself.
+- The method **should** support partial resource update, and the HTTP verb
+  **should** be `PATCH`.
+
+{% tab proto %}
+
+```proto
+rpc UpdateBook(UpdateBookRequest) returns (Book) {
+  option (google.api.http) = {
+    patch: "/v1/{book.path=publishers/*/books/*}"
+    body: "book"
+  };
+  option (google.api.method_signature) = "book,update_mask";
+}
+```
+
+- The resource's `path` field **should** map to the URI path.
+  - The `{resource}.path` field **should** be the only variable in the URI
+    path.
+- There **must** be a `body` key in the `google.api.http` annotation, and it
+  **must** map to the resource field in the request message.
+  - All remaining fields **should** map to URI query parameters.
+- There **should** be exactly one `google.api.method_signature` annotation,
+  with a value of `"{resource},update_mask"`.
+
+**Note:** Unlike the other four standard methods, the URI path here references
+a nested field (`book.path`) in the example. If the resource field has a word
+separator, `snake_case` is used.
+
+{% tab oas %}
+
+**Note:** OAS example not yet written.
+
+{% endtabs %}
+
+### Request message
+
+Update methods implement a common request message pattern:
+
+- The request message **must** contain a field for the resource.
+  - The name of this field **must** be the singular form of the resource's
+    name.
+- The request message **must not** contain any required fields other than those
+  described in this section, and **should not** contain other optional fields
+  except those described in this or another AEP.
```

Comment:
This is also proto-specific.

Reply:
Removed the word "message" - I think that's the only proto-specific part (but please unresolve if not).